iceoryx_hoofs/source/posix_wrapper/posix_access_rights.cpp

```diff
@@ -171,9 +168,7 @@ PosixUser::groupVector_t PosixUser::getGroups() const noexcept
     groupVector_t vec;
     for (int32_t i = 0; i < numGroups; ++i)
     {
-        /// NOLINTJUSTIFICATION @todo iox-#1614 will be fixed when upcoming cxx::array is used
-        /// NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
-        vec.emplace_back(PosixGroup(groups[i]));
+        vec.emplace_back(PosixGroup(groups[static_cast<uint64_t>(i)]));
```

Comment:
Ok, cast necessary because the index type of `operator[]` changed. Note that I am not sure anymore that having unsigned for this is a good idea for complicated reasons regarding type compatibility and optimization. More precisely it appears to be useful to have a signed version of the operator (and the unsigned one optional), to avoid precisely these issues for example (should not be added now, it is a more general issue and may warrant its own issue ticket).

I understand that such an index cannot be negative, but there is good reason to use signed index values by default unless the full unsigned range is required. We also require `numGroups` to be signed due to POSIX, so we cannot change the type of `i` to unsigned to avoid the cast.

Nothing to change unless, only confirm/deny.